src/order/lattice.lean

```diff
@@ -305,18 +417,73 @@ end
 
 end semilattice_inf
 
-/-! ### Lattices -/
+/-!
+### Lattices
+-/
 
 /-- A lattice is a join-semilattice which is also a meet-semilattice. -/
 class lattice (α : Type u) extends semilattice_sup α, semilattice_inf α
 
 instance (α) [lattice α] : lattice (order_dual α) :=
 { .. order_dual.semilattice_sup α, .. order_dual.semilattice_inf α }
 
+/-- The partial orders from `semilattice_sup_mk'` and `semilattice_inf_mk'` agree. -/
```

Comment:
I suppose this should mention that they agree given `sup_inf_self` and `inf_sup_self`

Reply:
Yeah, good idea:
```suggestion
/-- The partial orders from `semilattice_sup_mk'` and `semilattice_inf_mk'` agree
if `sup` and `inf` satisfy the lattice absorption laws `sup_inf_self` (`a ⊔ a ⊓ b = a`)
and `inf_sup_self` (`a ⊓ (a ⊔ b) = a`). -/
```